tests/test_preview.py

```diff
@@ -0,0 +1,706 @@
+    exp = Experiment(exp_name, exp_path=test_dir, launcher=test_launcher)
+
+    # Execute method for template rendering
+    output = previewrenderer.render(exp)
+
+    # Evaluate output
+    summary_lines = output.split("\n")
+    summary_lines = [item.replace("\t", "").strip() for item in summary_lines[-3:]]
+    assert 3 == len(summary_lines)
+    summary_dict = dict(row.split(": ") for row in summary_lines)
+    assert set(["Experiment", "Experiment Path", "Launcher"]).issubset(summary_dict)
+
+
+def test_experiment_preview_properties(test_dir, wlmutils):
+    """Test correct preview output properties for Experiment preview"""
+    # Prepare entities
+    test_launcher = wlmutils.get_test_launcher()
+    exp_name = "test_experiment_preview_properties"
+    exp = Experiment(exp_name, exp_path=test_dir, launcher=test_launcher)
+
+    # Execute method for template rendering
+    output = previewrenderer.render(exp)
+
+    # Evaluate output
+    summary_lines = output.split("\n")
+    summary_lines = [item.replace("\t", "").strip() for item in summary_lines[-3:]]
+    assert 3 == len(summary_lines)
+    summary_dict = dict(row.split(": ") for row in summary_lines)
+    assert exp.name == summary_dict["Experiment"]
+    assert exp.exp_path == summary_dict["Experiment Path"]
+    assert exp.launcher == summary_dict["Launcher"]
+
+
+def test_preview_output_format_html_to_file(test_dir, wlmutils):
+    """Test that an html file is rendered for Experiment preview"""
+    # Prepare entities
+    test_launcher = wlmutils.get_test_launcher()
+    exp_name = "test_preview_output_format_html"
+    exp = Experiment(exp_name, exp_path=test_dir, launcher=test_launcher)
+    filename = "test_preview_output_format_html.html"
+    path = pathlib.Path(test_dir) / filename
+
+    # Execute preview method
+    exp.preview(output_format="html", output_filename=str(path))
+
+    # Evaluate output
+    assert path.exists()
+    assert path.is_file()
+
+
+def test_preview_model_output_format(test_dir, wlmutils):
+    """
+    Test that an html file is rendered for Model preview
+    """
+    # Prepare entities
+    exp_name = "test_model_preview-html"
+    test_launcher = wlmutils.get_test_launcher()
+    exp = Experiment(exp_name, exp_path=test_dir, launcher=test_launcher)
+    model_params = {"port": 6379, "password": "unbreakable_password"}
+    rs1 = RunSettings("bash", "multi_tags_template.sh")
+    rs2 = exp.create_run_settings("echo", ["spam", "eggs"])
+
+    hello_world_model = exp.create_model(
+        "echo-hello", run_settings=rs1, params=model_params
+    )
+    spam_eggs_model = exp.create_model("echo-spam", run_settings=rs2)
+
+    filename = "test_model_preview_output_format.html"
+    path = pathlib.Path(test_dir) / filename
+
+    # Execute preview method
+    exp.preview(
+        hello_world_model,
+        spam_eggs_model,
+        output_format="html",
+        output_filename=str(path),
+    )
+
+    # Evaluate output
+    assert path.exists()
```

Comment:
1. Should we check that HTML tags exist or anything else about the content of the output file? I'd consider looking for something like `<html>` just to be sure it didn't fall back on a "default" format

2. What happens if i pass a path that is invalid? We might already have tests for the core functionality of preview like handling bad paths.

3. What happens if i give a output format that is valid but the template is incorrect. Ditto on potentially already covered.

Reply:
removing html from this PR. Will keep this note for future